compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1746,7 +1746,9 @@ impl<'tcx> TyCtxt<'tcx> {
                     MetadataKind::None
                 }
                 CrateType::Rlib => MetadataKind::Uncompressed,
-                CrateType::Dylib | CrateType::ProcMacro => MetadataKind::Compressed,
+                CrateType::Dylib | CrateType::ProcMacro | CrateType::Sdylib => {
+                    MetadataKind::Compressed
```

Comment:
`Sdylib`s aren't supposed to have metadata (well, in the usual unstable format at least).

Unless the interface libraries are also considered `CrateType::Sdylib`s during the build.
`CrateType::Sdylib` with and without `is_interface_build` behave quite differently.
Maybe they should be split into two crate types, or `CrateType::Sdylib` should be reset to `CrateType::Rlib` or something during the interface build? Not sure.

Or maybe `crate_type` and `is_interface_build` should just be passed together to all places where the distinction is necessary.